core/src/main/java/tc/oc/pgm/match/MatchPlayerImpl.java

```diff
@@ -199,6 +200,8 @@ public boolean canInteract() {
 
   @Override
   public boolean canSee(MatchPlayer other) {
+    @Nullable MatchPlayer spectatorTarget = this.getSpectatorTarget();
+    if (spectatorTarget != null && spectatorTarget.getId().equals(other.getId())) return true;
     if (!other.isVisible()) return false;
     if (other.isParticipating()) return true;
     if (other.isVanished() && !getBukkit().hasPermission(Permissions.VANISH)) return false;
```

Comment:
If someone you're spectating vanishes you still see them? nope

Don't return true in this case, rather ignore the specifically other not visible clausule, leave everything else as-is.

```suggestion
    @Nullable MatchPlayer spectatorTarget = this.getSpectatorTarget();
    boolean isSpectatorTarget = spectatorTarget != null && spectatorTarget.getId().equals(other.getId());
    if (!other.isVisible() && !isSpectatorTarget) return false;
    if (other.isParticipating()) return true;
    if (other.isVanished() && !getBukkit().hasPermission(Permissions.VANISH)) return false;
```